code/src/reducers/todos.js

```diff
@@ -0,0 +1,48 @@
+import { createSlice } from "@reduxjs/toolkit";
+import uniqid from "uniqid";
+
+const initialState = {
+  items: [],
+};
+
+export const todos = createSlice({
+  name: "todos",
+  initialState,
+  reducers: {
+    addItem: {
+      reducer(state, action) {
+        state.items = [...state.items, action.payload];
+      },
+      prepare(name) {
+        return {
+          payload: {
+            id: uniqid(),
+            name: name,
+            isCompleted: false,
+          },
+        };
+      },
```

Comment:
The structure of this reducer is a bit different than the ones below- is there a specific reason? Not really following here, since I can't figure out what "Prepare" is- is it a method? I'm curious! :)